daemon/set.go

```diff
@@ -0,0 +1,53 @@
+package daemon
+
+import (
+	"github.com/docker/docker/runconfig"
+)
+
+func (daemon *Daemon) ContainerSet(name string, hostConfig *runconfig.HostConfig) ([]string, error) {
+	warnings, err := daemon.verifyHostConfig(hostConfig)
+	if err != nil {
+		return warnings, err
+	}
+
+	container, err := daemon.Get(name)
+	if err != nil {
+		return warnings, err
+	}
+
+	daemon.updateResources(container, hostConfig)
+
+	if err := daemon.Set(container.command); err != nil {
+		return warnings, err
+	}
+
+	return warnings, nil
+}
+
+func (daemon *Daemon) updateResources(container *Container, hostConfig *runconfig.HostConfig) {
```

Comment:
We can also then check to see if the container is 'running' in container.Set() too.